packages/web3-providers-ws/src/index.ts

```diff
@@ -0,0 +1,391 @@
+import { EventEmitter } from 'events';
+import {
+	Web3BaseProvider,
+	Web3BaseProviderStatus,
+	JsonRpcId,
+	JsonRpcPayload,
+	JsonRpcResponse,
+	JsonRpcResult,
+	Web3BaseProviderCallback,
+} from 'web3-common';
+import { IMessageEvent, w3cwebsocket as W3WS } from 'websocket';
```

Comment:
It should not need to support to browser. Because browsers have native support for [Websocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API). You just need to use `isomorphic-ws` package and it will switch the context between native Websocket and `ws` implementation without any code change. 


The [comparison](https://npmcompare.com/compare/websocket,ws) between `ws` and `websocket` packages are not questionable. So we should stick to using `ws` package in our provider.

Reply:
by support it means: auto taken care/switched in browsers by lib. That's not done in ws, and its done in websocket. so as you mentioned for ws need switching via isomorphic . Opened [issue](https://github.com/ChainSafe/web3.js/issues/4515) for it,